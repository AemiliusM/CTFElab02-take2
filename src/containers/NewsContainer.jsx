import React, { Component } from 'react'
import Controls from '../components/articles/Controls';
import ArticleList from '../components/articles/ArticleList';
import { fetchArticles } from '../services/NewsApi';

export default class NewsContainer extends Component {
    state = {
        loading: true,
        articles: [],
        country: '',
    };

    async componentDidMount() {
        const articles = await fetchArticles('us');
        this.setState({ articles, loading:false });
    }

    handleCountryChange = (event) => {
        this.setState({ country: event.target.value })
    };

    handleSubmit = async (event) => {
        event.preventDefault();
        this.setState({ loading: true });
        const articles = await fetchArticles(this.state.country);
        this.setState({ articles, loading:false })
    };

    render() {
        const { loading, articles, country, handleCountryChange, handleSubmit } = this.state;

        return (
            <>
            {loading ? (
                <h1>Loading your Articles!</h1>
            ) : (
                <>
                <Controls
                country={country}
                onCountryChange={handleCountryChange}
                onSubmit={handleSubmit}
                />
                <ArticleList articles={articles} />
                </>
            )}
            </>
        )
    }
}