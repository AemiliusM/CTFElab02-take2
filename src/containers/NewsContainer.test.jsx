import {render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import NewsContainer from './NewsContainer';


describe('shows news articles', () => {
    it('should display a list of articles', async () => {
        render(<NewsContainer/>);

        screen.getByText('Loading your Articles!');

        const ul = await screen.findByRole('list', {name: 'articles'});
        expect(ul).toMatchSnapshot();

        const countryInput = await screen.findByLabelText('Country')
        userEvent.type(countryInput, 'ca');

        const submitButton = await screen.findByRole('button', {name: 'get-articles'});
        userEvent.click(submitButton);

        return waitFor(() => {
            const articles = screen.getAllByText('Title', 
            {exact: false});

            expect(articles).toHaveLength(5)
        });

    });
});