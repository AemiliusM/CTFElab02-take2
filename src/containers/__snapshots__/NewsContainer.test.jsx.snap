// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows news articles should display a list of articles 1`] = `
<ul
  aria-label="articles"
>
  <li>
    <div>
      <h1>
        Title: 
        Dignitaries gather at Washington National Cathedral for Colin Powell's funeral - CNN
      </h1>
      <span>
        Author: 
        Veronica Stracqualursi, CNN
      </span>
      <h3>
        Description: 
        The casket of the late Gen. Colin Powell has arrived at Washington National Cathedral, where former presidents, dignitaries and friends have gathered for the funeral service of the former secretary of state and chairman of the Joint Chiefs of Staff.
      </h3>
    </div>
  </li>
  <li>
    <div>
      <h1>
        Title: 
        Kate Beckinsale Liked A Post About Her Ex Pete Davidson's Charm After People Questioned His Rumored Relationship With Kim Kardashian Following Their Dinner Dates - BuzzFeed News
      </h1>
      <span>
        Author: 
        Leyla Mohammed
      </span>
      <h3>
        Description: 
        Kate's subtle comment came hours after Kim and Pete were spotted out to dinner for the second time in a week.
      </h3>
    </div>
  </li>
  <li>
    <div>
      <h1>
        Title: 
        The PS5 and Xbox Series X are available at Walmart today - The Verge
      </h1>
      <span>
        Author: 
        Alice Newcome-Beill, Antonio G. Di Benedetto
      </span>
      <h3>
        Description: 
        Walmart has announced restocks of the Playstation 5 and Xbox Series X, kicking off at 3PM ET / 12PM PT on its website. Availability is first come, first served, so youâ€™ll have to act fast.
      </h3>
    </div>
  </li>
  <li>
    <div>
      <h1>
        Title: 
        Bitcoin scam: Kroger confirms press release was faked - CNN
      </h1>
      <span>
        Author: 
        Reuters
      </span>
      <h3>
        Description: 
        Kroger said on Friday a press release touting the acceptance of bitcoin cash at its stores this holiday season is fraudulent, making it the second major retailer in recent weeks to get entangled in a crypto hoax.
      </h3>
    </div>
  </li>
  <li>
    <div>
      <h1>
        Title: 
        Russian diplomat found dead in Berlin was intelligence officer - The Guardian
      </h1>
      <span>
        Author: 
        Andrew Roth
      </span>
      <h3>
        Description: 
        German media identify man who apparently fell from window of Russian embassy as member of FSB
      </h3>
    </div>
  </li>
</ul>
`;
